src/transactions/ChangeTrustOpFrame.cpp

```diff
@@ -179,7 +187,8 @@ ChangeTrustOpFrame::doCheckValid(uint32_t ledgerVersion)
         }
     }
 
-    if (ledgerVersion > 15 && getSourceID() == getIssuer(mChangeTrust.line))
+    if (ledgerVersion > 15 && !isPoolLine &&
+        getSourceID() == getIssuer(mChangeTrust.line))
```

Comment:
I think it might be a better idea to use an `isIssuer` method that returns false for native and pool shares.

Reply:
Done